[ADD] pos: payment terminal documentation - adding Stripe (#2891)
odoo/documentation: content/applications/sales/point_of_sale/payment/stripe.rst

======
Stripe
======

Connecting a payment terminal allows you to offer a fluid payment flow to your customers and ease
the work of your cashiers.

Configuration
=============

Configure the payment method
----------------------------

Start by activating **Stripe** in the settings. From the **POS** dashboard, go to
:menuselection:`Configuration --> Settings --> Payment Terminals --> Check Stripe`.

Then, create the payment method:

- Click :menuselection:`Configuration --> Payment Methods --> Create`, and complete the
  :guilabel:`Method` field with your payment method's name;
- Set the :guilabel:`Journal` field as :guilabel:`Bank` and the :guilabel:`Use a Payment Terminal`
  field as :guilabel:`Stripe`;
- Enter your payment terminal serial number in the :guilabel:`Stripe Serial Number` field;
- Click on :guilabel:`Don't forget to complete Stripe connect before using this payment method.`

> .. image:: stripe/create-method-stripe.png

Reviewer: ```suggestion
.. image:: stripe/create-method-stripe.png
```
You forgot to :align: and :alt:

Reviewer: Still not addressed

.. note::
   - Do not check the :guilabel:`Identify Customer` field. Doing so would prevent unindentified
     customers from using Stripe.
   - The :guilabel:`Outstanding Account` and the :guilabel:`Intermediary Account` can stay empty to
     use the default accounts.
   - Find your payment terminal serial number under the device or on `Stripe's website <https://dashboard.stripe.com>`_.

Connect Stripe to Odoo
----------------------

Click on :guilabel:`Connect Stripe`. Doing so redirects you automatically to a configuration page.
Fill in all the information to create your Stripe account and link it with Odoo. Once the forms are
completed, the :guilabel:`Publishable Key` and the :guilabel:`Secret Key` are automatically loaded
in Odoo, and your terminal is ready to be used in a POS.

> .. image:: stripe/stripe-connect.png

Reviewer: ```suggestion
.. image:: stripe/stripe-connect.png
```
:align: center :alt: missing

Reviewer: Ditto

Author: it's just below

.. important::

>    - When you use **Stripe** exclusively in PoS, you only need the **Secret Key** to use your

Reviewer: ```suggestion
   - When you use **Stripe** exclusively in PoS, you only need the **Secret Key** to use your
```
PoS or POS, but only one on the page :)

Author: POS is the abbreviation for a point of sale (the concept), PoS here refers to the app Point of Sale

the work of your cashiers.

Configuration
=============

Configure the payment method
----------------------------

Start by activating **Stripe** in the settings. From the **POS** dashboard, go to
:menuselection:`Configuration --> Settings --> Payment Terminals --> Check Stripe`.

Then, create the payment method:

- Click :menuselection:`Configuration --> Payment Methods --> Create`, and complete the
  :guilabel:`Method` field with your payment method's name;
- Set the :guilabel:`Journal` field as :guilabel:`Bank` and the :guilabel:`Use a Payment Terminal`
  field as :guilabel:`Stripe`;
- Enter your payment terminal serial number in the :guilabel:`Stripe Serial Number` field;
- Click on :guilabel:`Don't forget to complete Stripe connect before using this payment method.`

.. image:: stripe/create-method-stripe.png
   :align: center
   :alt: payment method creation form

.. note::
   - Do not check the :guilabel:`Identify Customer` field. Doing so would prevent unindentified
     customers from using Stripe.
   - The :guilabel:`Outstanding Account` and the :guilabel:`Intermediary Account` can stay empty to
     use the default accounts.

>    - Find your payment terminal serial number under the device or on `Stripe's website <https://dashboard.stripe.com>`_.

Reviewer 2: ```suggestion
   - Find your payment terminal serial number under the device or on `Stripe's website
     <https://dashboard.stripe.com>`_.
```

Connect Stripe to Odoo
----------------------

Click on :guilabel:`Connect Stripe`. Doing so redirects you automatically to a configuration page.
Fill in all the information to create your Stripe account and link it with Odoo. Once the forms are

> completed, the :guilabel:`Publishable Key` and the :guilabel:`Secret Key` are automatically loaded
> in Odoo, and your terminal is ready to be used in a POS.

Reviewer 2: ```suggestion
completed, the API keys (:guilabel:`Publishable Key` and :guilabel:`Secret Key`) are automatically 
loaded in Odoo, and your terminal is ready to be used in a POS.
```

maybe "filled out" instead of "loaded"? Well, I don't know if it's better or not ^^'

> in Odoo, and your terminal is ready to be used in a POS.

Reviewer 2: Is this form only to create an account?
You assume that users don't have a stripe account yet. This paragraph doesn't say that they can login with their existing account.

> .. image:: stripe/stripe-connect.png

Reviewer 2: connection?
Stripe Connect is a product.
https://stripe.com/en-be/connect

   :align: center

>    :alt: stripe connect form

Reviewer 2: ```suggestion
   :alt: stripe connection form
```

that's not correct. It's a noun group so you want to write "connection"

.. important::
   - When you use **Stripe** exclusively in Point of Sale, you only need the **Secret Key** to use

>      your terminal.

Reviewer 2: is this important, though?
since the connection happens through a portal and the credentials are automatically filled, this isn't really important. At most, a side note.

>    - The :guilabel:`State` is inherent to **eCommerce** and can stay set as :guilabel:`Disabled`.

Reviewer 2: I don't think that this formulation conveys what you meant, with "inherent". Also, it is not only related to eCommerce, since payment providers can be used also in customer portals.

I think that it would be best to approach this from the payment provider perspective.

Author: I changed it, what do you think?

>    - For **On-Premise** users, the :guilabel:`Connect Stripe` button does not work. To retrieve the

Reviewer 2: "On-Premise users" may not be clear to users who don't know much about hosting types. Also, it's not about the users, but about the databases.
I would rather write 
`For databases hosted **On-Premise**`

Author: changed it

>      keys manually, go to `Stripe's website <https://dashboard.stripe.com>`_, log in, type

Reviewer 2: API keys
"log in to your stripe dashboard"
```suggestion
     API keys manually, log in to your `Stripe dashboard <https://dashboard.stripe.com>`_, type
```

     `API` in the search bar, and click :guilabel:`Developers > API`.

Configure the payment terminal
------------------------------

Swipe right on your payment terminal, click :menuselection:`Settings --> enter the admin PIN code

> --> Validate --> Network`, and select your network.

Reviewer 2: Don't use menuselection for a list of actions. It's only for menu navigation.
You can also make a numbered list if you prefer...

.. note::

>    - The device must be connected to a secured WI-FI.

Reviewer 2: "Wi-Fi network"
and not WI-FI

>    - Your **Odoo** database and payment terminal must share the same network.

Reviewer 2: why the bold for Odoo?

   - To access your payment terminal settings, you must enter the admin PIN code. By default, this
     code is `07139`.

Link the payment method to a POS
--------------------------------

To add a **payment method** to your point of sale, go to :menuselection:`Configuration -->
Settings`. Select the POS, go to :menuselection:`Payments --> Payment Methods`, and add your payment
method for **Stripe**.

Pay with a payment terminal
===========================

When processing a payment, select :guilabel:`Stripe` as the payment method. Check the amount and
click on :guilabel:`Send`. Once the payment is successful, the status changes to :guilabel:`Payment
Successful`. To cancel the payment request, click on :guilabel:`cancel`.

.. note::
   - | In case of connection issues between Odoo and the payment terminal, force the payment by
       clicking on :guilabel:`Force Done`, which allows you to validate the order.
     | This option is only available after receiving an error message informing you that the
       connection failed.

>    - The terminal must have at least 10% battery to use it.

Reviewer 2: "battery level", not just "battery".
Otherwise, it's 10% of the physical battery ^^

   - The device does not work for payments under €0.50.

Troubleshooting
===============

You cannot find the payment terminal in your Stripe account.

> ------------------------------------------------------------

Reviewer 2: payment terminal unavailable in your stripe account
?

https://www.odoo.com/documentation/master/contributing/documentation/content_guidelines.html#titles-and-headings

If you cannot find a payment terminal, you need to add it manually.

> #. Go to `Stripe's website <https://dashboard.stripe.com>`_, log in, and go to

Reviewer 2: same suggestion as above.

Reviewer 2: you didn't apply the same thing as line 60
```suggestion
#. Log in to your `Stripe dashboard <https://dashboard.stripe.com>`_, and go to
```

Activate **Stripe** in the settings by going to :menuselection:`Point of Sale --> Configuration -->
Settings --> Payment Terminals` and enabling :guilabel:`Stripe`.

Then, create the payment method:

- Go to :menuselection:`Point of Sale --> Configuration --> Payment Methods`, click
  :guilabel:`Create`, and complete the :guilabel:`Method` field with your payment method's name;
- Set the :guilabel:`Journal` field as :guilabel:`Bank` and the :guilabel:`Use a Payment Terminal`
  field as :guilabel:`Stripe`;
- Enter your payment terminal serial number in the :guilabel:`Stripe Serial Number` field;
- Click :guilabel:`Don't forget to complete Stripe connect before using this payment method.`

.. image:: stripe/create-method-stripe.png
   :align: center
   :alt: payment method creation form

.. note::
   - Click :guilabel:`Identify Customer` to allow this payment method **exclusively** for identified
     customers. For any unidentified customers to be able to pay with Stripe, leave the
     :guilabel:`Identify Customer` field unchecked.
   - The :guilabel:`Outstanding Account` and the :guilabel:`Intermediary Account` can stay empty to
     use the default accounts.

>    - Find your payment terminal serial number under the device or on `Stripe's website

Reviewer 2: dashboard, like you wrote at line 60
```suggestion
   - Find your payment terminal serial number under the device or on `Stripe's dashboard
```

     <https://dashboard.stripe.com>`_.

Connect Stripe to Odoo
----------------------

Click :guilabel:`Connect Stripe`. Doing so redirects you automatically to a configuration page.
Fill in all the information to create your Stripe account and link it with Odoo. Once the forms are
completed, the API keys (:guilabel:`Publishable Key` and :guilabel:`Secret Key`) are automatically

> loaded in Odoo, and your terminal is ready to be used in a POS.

Reviewer 3: That's not true. You must copy them from your Stripe Dashboard first.

--------------------------------

To add a **payment method** to your point of sale, go to :menuselection:`Point of Sale -->
Configuration --> Settings`. Select the POS, scroll down to the :guilabel:`Payments` section, and
add your payment method for **Stripe** in the :guilabel:`Payment Methods` field.

Pay with a payment terminal
===========================

When processing a payment, select :guilabel:`Stripe` as the payment method. Check the amount and
click :guilabel:`Send`. Once the payment is successful, the status changes to :guilabel:`Payment
Successful`. To cancel the payment request, click :guilabel:`cancel`.

.. note::
   - | In case of connection issues between Odoo and the payment terminal, force the payment by
       clicking on :guilabel:`Force Done`, which allows you to validate the order.
     | This option is only available after receiving an error message informing you that the
       connection failed.
   - The terminal must have at least 10% battery level to use it.
   - The device does not work for payments under €0.50.

Troubleshooting
===============

Payment terminal unavailable in your stripe account
---------------------------------------------------

> If the payment terminal is unavailable in your stripe account, you must add it manually:

Reviewer 2: ```suggestion
If the payment terminal is unavailable in your Stripe account, you must add it manually:
```

Connect Stripe to Odoo
----------------------

Click on :guilabel:`Connect Stripe`. Doing so redirects you automatically to a configuration page.
Fill in all the information to create your Stripe account and link it with Odoo. Once the forms are
completed, the :guilabel:`Publishable Key` and the :guilabel:`Secret Key` are automatically loaded
in Odoo, and your terminal is ready to be used in a POS.

.. image:: stripe/stripe-connect.png

.. important::
   - When you use **Stripe** exclusively in PoS, you only need the **Secret Key** to use your
     terminal.
   - The :guilabel:`State` is inherent to **eCommerce** and can stay set as :guilabel:`Disabled`.
   - For **On-Premise** users, the :guilabel:`Connect Stripe` button does not work. To retrieve the
     keys manually, go to `Stripe's website <https://dashboard.stripe.com/register>`_, log in, type
     `API` in the search bar, and click :guilabel:`Developers > API`.

Configure the payment terminal
------------------------------

Swipe right on your payment terminal, click :menuselection:`Settings --> enter the admin PIN code
--> Validate --> Network`, and select your network.

.. note::

>    - The device must be connected to a secured WIFI.

Reviewer: ```suggestion
   - The device must be connected to a secured Wi-Fi.
```
It is actually spelled this way

   - Your **Odoo** database and payment terminal must share the same network.
   - To access your payment terminal settings, you must enter the admin PIN code. By default, this
     code is `07139`.

Add a new payment method
------------------------

To add a new **payment method** to your point of sale, go to :menuselection:`Configuration -->
Settings`. Select the POS, go to :menuselection:`Payments --> Payment Methods`, and add your payment

> method for **Stripe**.

Reviewer 4: Hi @LoredanaLrpz, nice one, I like it !

Just one suggestion here, "Add a new payment method" sounds redundant with the previous "Configure a payment method" IMHO, although those sections speaks well about different things.

To avoid confusion, why not:
* renaming this section "Link the new payment method"
* and start with "To add the newly configured **payment method** to your point of sale, ...."

What do you think ? Sorry if I sound peaky :)

Pay with a payment terminal
===========================

When processing a payment, select :guilabel:`Stripe` as the payment method. Check the amount and
click on :guilabel:`Send`. Once the payment is successful, the status changes to :guilabel:`Payment
Successful`. To cancel the payment request, click on :guilabel:`cancel`.

.. note::

>    - | In case of connexion issues between Odoo and the payment terminal, force the payment by

Reviewer: ```suggestion
   - | In case of connection issues between Odoo and the payment terminal, force the payment by
```

       clicking on :guilabel:`Force Done`, which allows you to validate the order.

>      | This option is only available after receiving an error message informing you that the

Reviewer: There are weird gaps between the lines in the note. This option is on a new line but without any spaces between it and the previous line, but the t
![2022-11-15_09-03](https://user-images.githubusercontent.com/104567387/201863370-7cc79719-3400-43b0-9769-3011f598ec07.png)
wo others have a gap between each.

       connection failed.

>    -   The terminal must have at least 10% battery to use it.

Reviewer: ```suggestion
   -   The terminal must have at least 10% battery to be able to use it.
```

>    -   The device does not work for payments under €0.50.

Reviewer: ```suggestion
   -   The device does not work for payments under €0.50.
```
Does Stripe only operate in €?

Author: it accepts (according to their website) +135 currencies, but I don't know how to phrase that without the euro symbol. Good catch though, I'm asking Dama about that
```suggestion
   -   The device does not work for amounts under 0.50.
```

Reviewer: "does not work under a certain minimum amount" ?

Troubleshooting
===============

You cannot find the payment terminal in your Stripe account.
------------------------------------------------------------

> Create a terminal on Stripe's website by clicking on the :guilabel:`New` button.

Reviewer: ```suggestion
Create a terminal on Stripe's website by clicking on the :guilabel:`New` button.
```
Where  ?

Author: I changed it as it was really unclear

======
Stripe
======

Connecting a payment terminal allows you to offer a fluid payment flow to your customers and ease
the work of your cashiers.

Configuration
=============

Create your Stripe account
--------------------------

> Start by creating your **Stripe** account on `Stripe's website <https://dashboard.stripe.com/register>`_.

Author: I've tried this flow from PoS by clicking directly on "don't forget..." then "connect stripe," and it works without having to go to "Invoicing" so I'll stick to that.
You mentioned there is no need to create a stripe account beforehand, and you're right. Runbot was the problem, but in the saas trial version, all the information is asked directly :)

.. seealso::

>    - `Stripe Documentation <https://stripe.com/docs/payments?payments=popular>`_.

Author: I took it off

connect before using this payment method.`

.. image:: stripe/create-method-stripe.png

.. note::
   - Find your payment terminal serial number under the device or on `Stripe's website <https://dashboard.stripe.com/register>`_.

Connect Stripe to Odoo
++++++++++++++++++++++

Click on :guilabel:`Connect Stripe`. You are redirected to a configuration page. Follow the
instructions to connect your terminal.

.. image:: stripe/stripe-connect.png

Once the process is over, the :guilabel:`Publishable Key` and the :guilabel:`Secret Key` are
automatically completed.

.. important::
   - When you use **Stripe** exclusively in PoS, you only need the **Secret Key** to use your
     terminal.
   - The :guilabel:`State` is inherent to **eCommerce** and can stay set as :guilabel:`Disabled`.
   - For **On-Premise** users, the :guilabel:`Connect Stripe` button does not work. To retrieve the
     keys manually, go to `Stripe's website <https://dashboard.stripe.com/register>`_, log in, type
     `API` in the search bar, and click :guilabel:`Developers > API`.

Configure the payment terminal
++++++++++++++++++++++++++++++

> Swipe right on your payment terminal, click :menuselection:`Settings --> enter 07319 --> Validate

Author: You're absolutely right, I changed it

--> Network`, and select your network.

.. note::
   - The device must be connected to a secured WIFI.
   - Your **Odoo** database and payment terminal must share the same network.
   - To access your payment terminal settings, you must enter the admin pin code `07139`.

Add a new payment method

> ========================

Reviewer 4: I've the feeling this is repeating the previous section **Configure the payment method**

Author: When you are done creating the payment method, you still have to go back and add It to the pos, otherwise you cannot use it
